Python/pythonrun.c

```diff
@@ -39,17 +39,27 @@
 #include "windows.h"
 #endif
 
+_Py_IDENTIFIER(__main__);
+_Py_IDENTIFIER(__module__);
 _Py_IDENTIFIER(builtins);
+_Py_IDENTIFIER(code);
+_Py_IDENTIFIER(encoding);
 _Py_IDENTIFIER(excepthook);
+_Py_IDENTIFIER(filename);
 _Py_IDENTIFIER(flush);
 _Py_IDENTIFIER(last_traceback);
 _Py_IDENTIFIER(last_type);
 _Py_IDENTIFIER(last_value);
+_Py_IDENTIFIER(lineno);
+_Py_IDENTIFIER(msg);
```

Comment:
could the _Py_IDENTIFIER macro be modified such that it _only_ works from the global scope so that these don't creep back into function bodies?